ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+from ansible_collections.arista.avd.plugins.module_utils.utils import get
+from ansible_collections.arista.avd.plugins.filter.snmp_hash import hash_passphrase
+from hashlib import sha1
+
+
+class AVDStructConfig(AVDFacts):
+
+    def __init__(self, hostvars=None):
+        self._hash_passphrase = hash_passphrase
+        super().__init__(hostvars)
+
+    @cached_property
+    def _mgmt_interface_vrf(self):
+        return get(self._hostvars, 'mgmt_interface_vrf')
+
+    @cached_property
+    def _mgmt_gateway(self):
+        return get(self._hostvars, 'mgmt_gateway')
+
+    @cached_property
+    def _switch_data(self):
+        return get(self._hostvars, 'switch')
+
+    @cached_property
+    def _internal_vlan_order(self):
+        return get(self._hostvars, 'internal_vlan_order')
+
+    @cached_property
+    def _platform_settings(self):
+        return get(self._switch_data, 'platform_settings')
+
+    @cached_property
+    def _mgmt_ip(self):
+        return get(self._switch_data, 'mgmt_ip')
+
+    @cached_property
+    def _hostname(self):
+        """
+        switch.hostname fact set based on inventory_hostname variable
+        """
+        return get(self._hostvars, "inventory_hostname", required=True)
+
+    @cached_property
+    def router_bgp(self):
+        bgp_as = get(self._switch_data, "bgp_as")
+        if bgp_as is not None:
+            bgp_defaults = get(self._switch_data, 'bgp_defaults')
+            bgp_maximum_paths = get(self._hostvars, 'bgp_maximum_paths')
+            bgp_ecmp = get(self._hostvars, 'bgp_ecmp')
+            if bgp_maximum_paths is not None:
+                max_paths_str = f"maximum-paths {bgp_maximum_paths}"
+                if get(self._hostvars, 'bgp_ecmp') is not None:
+                    max_paths_str += f" ecmp {bgp_ecmp}"
+
+                bgp_defaults.append(max_paths_str)
+
+            return {
+                'as': get(self._switch_data, 'bgp_as'),
+                'router_id': get(self._switch_data, 'router_id'),
+                'bgp_defaults': bgp_defaults,
+            }
+        return None
+
+    @cached_property
+    def static_routes(self):
+        if self._mgmt_gateway is not None:
+            static_routes = []
+            mgmt_destination_networks = get(self._hostvars, 'mgmt_destination_network')
+            if mgmt_destination_networks is not None:
+                for mgmt_destination_network in self._mgmt_destination_networks:
+                    static_routes.append({'vrf': self._mgmt_interface_vrf,
+                                          'destination_address_prefix': mgmt_destination_network,
+                                          'gateway': self._mgmt_gateway})
+            else:
+                static_routes.append({'vrf': self._mgmt_interface_vrf,
+                                      'destination_address_prefix': '0.0.0.0/0',
+                                      'gateway': self._mgmt_gateway})
+            return static_routes
+        return None
```

Comment:
is there a reason to return `None` vs `[]` depending upon which variable is not present?

```suggestion
        if self._mgmt_gateway is None:
            return None
        static_routes = []
        mgmt_destination_networks = get(self._hostvars, 'mgmt_destination_network')
        if mgmt_destination_networks is not None:
            for mgmt_destination_network in self._mgmt_destination_networks:
                static_routes.append({'vrf': self._mgmt_interface_vrf,
                                      'destination_address_prefix': mgmt_destination_network,
                                      'gateway': self._mgmt_gateway})
        else:
            static_routes.append({'vrf': self._mgmt_interface_vrf,
                                  'destination_address_prefix': '0.0.0.0/0',
                                  'gateway': self._mgmt_gateway})
        return static_routes
```

Reply:
Only situation where we would return [] is when mgmt_destination_networks is []. Only downside is that it will trigger some empty documentation tables in eos_cli_config_gen.

Reply:
yeah it was like this before the suggestion as well - maybe we can test on the empty list at the end and return None then